tests/func-tests/monitoring_test.go

```diff
@@ -283,3 +340,96 @@ func getPrometheusURL(ctx context.Context, cli rest.Interface) string {
 
 	return fmt.Sprintf("https://%s", route.Spec.Host)
 }
+
+const tempRouteName = "prom-route"
+
+// createTempRoute creates a route to the HCO prometheus endpoint, to allow reading the metrics.
+func createTempRoute(ctx context.Context, cli client.Client) error {
+	route := &openshiftroutev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tempRouteName,
+			Namespace: tests.InstallNamespace,
+		},
+		Spec: openshiftroutev1.RouteSpec{
+			Port: &openshiftroutev1.RoutePort{
+				TargetPort: intstr.FromString("http-metrics"),
+			},
+			TLS: &openshiftroutev1.TLSConfig{
+				Termination:                   openshiftroutev1.TLSTerminationEdge,
+				InsecureEdgeTerminationPolicy: openshiftroutev1.InsecureEdgeTerminationPolicyRedirect,
+			},
+			To: openshiftroutev1.RouteTargetReference{
+				Kind:   "Service",
+				Name:   "kubevirt-hyperconverged-operator-metrics",
+				Weight: ptr.To[int32](100),
+			},
+			WildcardPolicy: openshiftroutev1.WildcardPolicyNone,
+		},
+	}
+
+	return cli.Create(ctx, route)
+}
+
+func deleteTempRoute(ctx context.Context, cli client.Client) error {
+	route := &openshiftroutev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tempRouteName,
+			Namespace: tests.InstallNamespace,
+		},
+	}
+	return cli.Delete(ctx, route)
+}
+
+func getTempRouteHost(ctx context.Context, cli client.Client) (string, error) {
+	route := &openshiftroutev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tempRouteName,
+			Namespace: tests.InstallNamespace,
+		},
+	}
+	err := cli.Get(ctx, client.ObjectKeyFromObject(route), route)
+	if err != nil {
+		return "", fmt.Errorf("failed to read the temp router; %w", err)
+	}
+
+	if len(route.Status.Ingress) == 0 {
+		return "", fmt.Errorf("failed to read the temp route status")
+	}
+
+	return route.Status.Ingress[0].Host, nil
+}
+
+func getHCOMetric(ctx context.Context, url, query string) (float64, error) {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	httpClient := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
can we extract the CAFile from the `rest.Config` object used to initialize the `client.Client` we are using to talk with the APIServer and trust this?

Reply:
done